x-pack/plugin/security/cli/src/main/java/org/elasticsearch/xpack/security/cli/AutoConfigInitialNode.java

```diff
@@ -0,0 +1,352 @@
+                return;
+            }
+            // Silently skipping security auto configuration because TLS is already configured
+            if (false == env.settings().getByPrefix(XPackSettings.TRANSPORT_SSL_PREFIX).isEmpty() ||
+                    false == env.settings().getByPrefix(XPackSettings.HTTP_SSL_PREFIX).isEmpty()) {
+                // zero validation for the TLS settings as well, let the node bootup do its thing
+                terminal.println(Terminal.Verbosity.VERBOSE,
+                        "Skipping security auto configuration because it appears that TLS is already configured.");
+                return;
+            }
+            // Silently skipping security auto configuration because file realm is configured
+            // Auto-configuration only requires that the file realm be enabled, and, for optimal experience,
+            // also be the first in the chain.
+            if (false == env.settings().getByPrefix(RealmSettings.realmSettingPrefix(FileRealmSettings.TYPE)).isEmpty()) {
+                terminal.println(Terminal.Verbosity.VERBOSE,
+                        "Skipping security auto configuration because it appears that a file-based realm is already configured.");
+                return;
+            }
+        }
+        final Path ymlPath = env.configFile().resolve("elasticsearch.yml");
+        // The node's yml configuration is mounted read-only, which is a sign that the configuration mustn't change.
+        // Inform that auto-configuration will not run.
+        if (false == Files.isWritable(ymlPath)) {
+            terminal.println(Terminal.Verbosity.NORMAL, String.format(Locale.ROOT, "Skipping security auto configuration because " +
+                    "the configuration file [%s] is not writable", ymlPath));
+            return;
+        }
+        // Best effort determine if the node's keystore is writable.
+        // Inform that auto-configuration will not run if keystore cannot be updated.
+        final Path keystorePath = env.configFile().resolve(KeyStoreWrapper.KEYSTORE_FILENAME);
+        if (Files.exists(keystorePath) && false == Files.isWritable(keystorePath)) {
+            terminal.println(Terminal.Verbosity.NORMAL, String.format(Locale.ROOT, "Skipping security auto configuration because " +
+                    "the node keystore file [%s] is not writable", keystorePath));
+            return;
+        }
+
+        // TODO round/truncate to day, as it is easier on the tests
+        //Rounding rounding = new Rounding.Builder(Rounding.DateTimeUnit.DAY_OF_MONTH).timeZone(ZoneOffset.UTC).build();
+        //ZonedDateTime autoConfigDate = ZonedDateTime.now(ZoneOffset.UTC).truncatedTo(ChronoUnit.DAYS);
+        final ZonedDateTime autoConfigDate = ZonedDateTime.now(ZoneOffset.UTC);
+        final String instantAutoConfigName = "auto_generated_on_" + autoConfigDate.toInstant().getEpochSecond();
```

Comment:
Is there any use/value for a variable dir name ? I have nothing against it from a technical perspective, but we might want to point to this dir in our docs ( i.e. there you can find tha CA certificate for HTTPS to configure your client to trust ES's certificate) and it would be easier to point to a hardcoded path i.e. $ES_PATH_CONF/auto_generated_cert

Reply:
> Is there any use/value for a variable dir name ?

It is just my slight preference that auto generated files have some path part that is unique for a group. In this way one can tell that a group of files have been generated together and that they should be used together. If the name is fixed and the auto-configuration outputs can mix and there's no easy way to untangle them because they are password protected.

If you think we can accommodate a variable name in the docs, I would prefer we keep the variable name.
What do you think?

Reply:
No strong opinions here. I see the value in what you're saying. On the other hand, I don't see how auto-configuration outputs can mix and we/users can always check the last modified timestamp if we need to. Fine to keep this and we'll figure out a way to present it in a non confusing way in the docs :+1: